pandas/tests/indexing/test_multiindex.py

```diff
@@ -198,6 +202,27 @@ def test_loc_getitem_array(self):
         result = x.loc[scalar]
         tm.assert_series_equal(result, expected)
 
+    def test_loc_generator(self):
```

Comment:
This PR removes some broken code which prevented ``__getitem__(generator)`` to work, so yes, strictly speaking it fixes it, but this was a totally unintended consequence. I actually made sure to keep the tests for generators in two independent commits, which you can drop if you don't think they are useful. But there was already [at least one test](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/series/test_indexing.py#L270) on indexing a ``Series`` with generators, I just expanded a bit. No idea about indexing ``Index``es.